Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -1,21 +1,254 @@
+    //   refactor their implementation later on.
+  }
+  /// A measurement can only be constructed with valid parameters.
+  FixedSizeMeasurement() = delete;
+  FixedSizeMeasurement(const FixedSizeMeasurement&) = default;
+  FixedSizeMeasurement(FixedSizeMeasurement&&) = default;
+  ~FixedSizeMeasurement() = default;
+  FixedSizeMeasurement& operator=(const FixedSizeMeasurement&) = default;
+  FixedSizeMeasurement& operator=(FixedSizeMeasurement&&) = default;
+
+  /// Source link that connects to the underlying detector readout.
+  const Acts::SourceLink& sourceLink() const { return m_source; }
+
+  /// Number of measured parameters.
+  static constexpr std::size_t size() { return kSize; }
+
+  /// Check if a specific parameter is part of this measurement.
+  bool contains(indices_t i) const { return m_subspace.contains(i); }
+
+  /// The measurement indices
+  constexpr std::array<indices_t, kSize> indices() const {
+    std::array<uint8_t, kSize> subInds = m_subspace.indices();
+    std::array<indices_t, kSize> inds{};
+    for (std::size_t i = 0; i < kSize; i++) {
+      inds[i] = static_cast<indices_t>(subInds[i]);
+    }
+    return inds;
+  }
+
+  /// Measured parameters values.
+  const ParametersVector& parameters() const { return m_params; }
+
+  /// Measured parameters covariance.
+  const CovarianceMatrix& covariance() const { return m_cov; }
+
+  /// Projection matrix from the full space into the measured subspace.
+  ProjectionMatrix projector() const {
+    return m_subspace.template projector<Scalar>();
+  }
+
+  /// Expansion matrix from the measured subspace into the full space.
+  ///
+  /// This is equivalent to the transpose of the projection matrix only in the
+  /// case of a trivial projection matrix. While this is the case here, it is
+  /// still recommended to use the expansion matrix directly in cases where it
+  /// is explicitly used.
+  ExpansionMatrix expander() const {
+    return m_subspace.template expander<Scalar>();
+  }
+
+  /// Compute residuals in the measured subspace.
+  ///
+  /// @param reference Reference parameters in the full space.
+  ///
+  /// This computes the difference `measured - reference` taking into account
+  /// the allowed parameter ranges. Only the reference values in the measured
+  /// subspace are used for the computation.
+  ParametersVector residuals(const FullParametersVector& reference) const {
+    ParametersVector res = ParametersVector::Zero();
+    Acts::detail::calculateResiduals(static_cast<indices_t>(kSize),
+                                     m_subspace.indices(), reference, m_params,
+                                     res);
+    return res;
+  }
+
+  std::ostream& operator<<(std::ostream& os) const {
+    Acts::detail::printMeasurement(os, static_cast<indices_t>(kSize),
+                                   m_subspace.indices().data(), m_params.data(),
+                                   m_cov.data());
+    return os;
+  }
+
+ private:
+  Acts::SourceLink m_source;
```

Comment:
Should this even have a source link itself? Are we accessing this at all?

Reply:
I think we can remove this

Reply:
This seems to be used in some writer. Maybe we can push this to a second iteration if necessary?